module/Finna/src/Finna/View/Helper/Root/Record.php

```diff
@@ -917,4 +917,53 @@ public function getAuthoritySummary($onAuthorityPage = false)
             ['summary' => $summary, 'driver' => $this->driver]
         );
     }
+
+    /**
+     * Returns additional information related to external material links.
+     *
+     * @return string
+     */
+    public function getExternalLinkAdditionalInfo()
+    {
+        $translator = $this->getView()->plugin('translate');
+        $externalLinkText = $translator('external_link');
+        switch ($this->driver->getDataSource()) {
+        case 'aoe':
+            $source = ' aoe.fi';
+            break;
+        default:
+            $source = '';
+        }
+        return '(' . $externalLinkText . $source . ')';
+    }
+
+    /**
+     * Returns a data source specific material disclaimer.
+     *
+     * @return string
+     */
+    public function getMaterialDisclaimer()
+    {
+        return $this->renderTemplate(
```

Comment:
Tässä pitäisi catchata Laminas\View\Exception\RuntimeException. Muuten kaatuu, jos templaattia ei ole. Näin kävisi esim. jos datasourcena olisi aoe_test...

Reply:
Vaihtoehtoisesti pitäisi olla DefaultRecord:lla tyhjä material-disclaimer.phtml ja SolrLrmi:llä sellainen, joka tarkistaa driverin data sourcen ja tulostaa sen perusteella jotain.